examples/vision/segformer.py

```diff
@@ -0,0 +1,288 @@
+then used to guide the training process.
+
+With the compiled, we can proceed and call `fit()` on it to begin the fine-tuning
+process! 
+"""
+
+"""
+## Prediction callback to monitor training progress
+
+It helps us to visualize some sample predictions when the model is being fine-tuned,
+thereby helping us to monitor the progress of the model. This callback is inspired from
+[this tutorial](https://www.tensorflow.org/tutorials/images/segmentation). 
+"""
+
+from IPython.display import clear_output
+
+
+def create_mask(pred_mask):
+    pred_mask = tf.math.argmax(pred_mask, axis=1)
+    pred_mask = tf.expand_dims(pred_mask, -1)
+    return pred_mask[0]
+
+
+def show_predictions(dataset=None, num=1):
+    if dataset:
+        for sample in dataset.take(num):
+            images, masks = sample["pixel_values"], sample["labels"]
+            masks = tf.expand_dims(masks, -1)
+            pred_masks = model.predict(images).logits
+            images = tf.transpose(images, (0, 2, 3, 1))
+            display([images[0], masks[0], create_mask(pred_masks)])
+    else:
+        display(
+            [
+                sample_image,
+                sample_mask,
+                create_mask(model.predict(tf.expand_dims(sample_image, 0))),
+            ]
+        )
+
+
+class DisplayCallback(tf.keras.callbacks.Callback):
+    def __init__(self, dataset, **kwargs):
+        super().__init__(**kwargs)
+        self.dataset = dataset
+
+    def on_epoch_end(self, epoch, logs=None):
+        clear_output(wait=True)
+        show_predictions(self.dataset)
+        print("\nSample Prediction after epoch {}\n".format(epoch + 1))
+
+
+"""
+## Train model
+"""
+
+# Increase the number of epochs if the results are not of expected quality.
+epochs = 5
+
+history = model.fit(
+    train_ds,
+    validation_data=test_ds,
+    callbacks=[DisplayCallback(test_ds)],
+    epochs=epochs,
+)
+
+"""
+## Inference
+
+Perform inference on a few samples from the test set. 
+"""
+
+show_predictions(test_ds, 5)
+
+"""
+## Conclusion
+
+In this example, we learned how to fine-tune a SegFormer model variant on a custom
+dataset for the task of semantic segmentation. In the interest of brevity, the example
+was kept short. However, there are a couple of things, you can further try out:
+
+* Incorporate data augmentation to potentially improve the results. 
+* Use a larger SegFormer model checkpoint to see how the results are affected. 
+* Push the fine-tuned model to the Hugging Face for sharing with the community easily.
+You can do so just by doing `model.push_to_hub("your-username/your-awesome-model")`.
+And then you can load the model by doing
+`TFSegformerForSemanticSegmentation.from_pretrained("your-username/your-awesome-model"`).
+[Here](https://github.com/huggingface/notebooks/blob/main/examples/semantic_segmentation-tf.ipynb)
+is an end-to-end example if you're looking for a reference.
+* If you'd rather want to push the model checkpoints to the Hub as the model is being
```

Comment:
nit: I'd say "If you'd rather {do task}" - don't ask me to explain the grammar rule 🙈 

```suggestion
* If you'd rather push the model checkpoints to the Hub as the model is being
```